.github/workflows/main.yaml

```diff
@@ -9,8 +9,22 @@ jobs:
   gomod:
     runs-on: ubuntu-20.04
     steps:
+    - id: go-cache-paths
```

Comment:
It looks like this workflow (and maybe others as well) is not using the [setup-go](https://github.com/actions/setup-go) action. That action utilizes caching by default and does "the right thing" in terms of assigning a key to the cache, etc. It probably makes sense to update the workflows to use the `setup-go` action.

Also, with caching enabled, the need for uploading some of the artifacts below goes away.